zap/src/main/java/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -215,33 +218,61 @@ public void setMessage(HttpMessage msg) {
     @Override
     public String setParameter(
             HttpMessage msg, NameValuePair originalPair, String name, String value) {
-        return setParameter(msg, originalPair, value);
+        return setParameter(msg, Arrays.asList(originalPair.getPosition()), Arrays.asList(value));
     }
 
     @Override
     public String setEscapedParameter(
             HttpMessage msg, NameValuePair originalPair, String name, String value) {
-        return setParameter(msg, originalPair, value);
+        return setParameter(msg, Arrays.asList(originalPair.getPosition()), Arrays.asList(value));
```

Comment:
Could use `Collections.singletonList` instead.

Reply:
thanks for mentioning it.